includes/commerce_pos.payment.inc

```diff
@@ -40,11 +48,12 @@ function commerce_pos_payment(&$form, &$form_state, $transaction_type) {
 
   form_load_include($form_state, 'inc', 'commerce_pos', 'includes/commerce_pos.common');
   commerce_pos_payment_ajax_check($form, $form_state);
+  dpm($form_state);
 
   $return = $transaction_type == CommercePosService::TRANSACTION_TYPE_RETURN;
   $payment_options = commerce_pos_get_payment_options();
   $form['#theme'] = 'commerce_pos_payment';
-  $form['#redirect_key'] = $return ? 'return' : 'sale';
+  $form['#redirect_key'] = $return ? 'return' : ($transaction_type == CommercePosService::TRANSACTION_TYPE_EXCHANGE ? 'exchange' : 'sale');
```

Comment:
Yeah, I agree. I just don't know why they did $return ? 'return' : 'sale' if the transaction_type already holds one of those values. I just didn't want to mess with their logic. But, since you also bring it up, I'm going to restructure it.